proxy/secret.go

```diff
@@ -30,161 +30,123 @@ func (c *Client) GetSecretList(ctx context.Context, namespace string) ([]types.S
 	getRequest, err := c.newRequest(http.MethodGet, secretPath, nil)
 
 	if err != nil {
-		return nil, fmt.Errorf("cannot connect to OpenFaaS on URL: %s", c.GatewayURL.String())
+		return nil, nil, fmt.Errorf("cannot connect to OpenFaaS on URL: %s", c.GatewayURL.String())
 	}
 
 	res, err := c.doRequest(ctx, getRequest)
 	if err != nil {
-		return nil, fmt.Errorf("cannot connect to OpenFaaS on URL: %s", c.GatewayURL.String())
+		return nil, res, fmt.Errorf("cannot connect to OpenFaaS on URL: %s", c.GatewayURL.String())
 	}
 
 	if res.Body != nil {
 		defer res.Body.Close()
 	}
 
+	bytesOut, err := ioutil.ReadAll(res.Body)
```

Comment:
have we upgraded to 1.16 yet? they moved most of the ioutil packages, this is now in `io.ReadAll`, how many Go versions do we want to support?

Reply:
Also, in theory, Body could be nil here right? I don't think `ReadAll` handles a `nil` reader so in some odd edge cases we would get a panic here. We didn't hndle it before, but perhaps it something we want to improve here and in the other methods?

https://golang.org/src/io/io.go?s=19249:19287#L616